src/api-client.ts

```diff
@@ -93,53 +110,84 @@ export class ApiClient {
         }
       }
 
-      return res.data.data.attributes.credentials
+      return res.result.data.attributes.credentials
     } catch (error: unknown) {
-      if (axios.isAxiosError(error)) {
-        const err = error
-        if (err.response) {
-          throw new CredentialFetchingError(
-            `fetching credentials: received code ${err.response
-              ?.status}: ${JSON.stringify(err.response?.data)}`
-          )
-        } else {
-          throw new CredentialFetchingError(
-            `fetching credentials: ${err.message}`
-          )
-        }
-      } else {
+      if (error instanceof CredentialFetchingError) {
+        throw error
+      } else if (error instanceof httpClient.HttpClientError) {
+        throw new CredentialFetchingError(
+          `fetching credentials: unexpected status code: ${error.statusCode}: ${error.message}`
+        )
+      } else if (error instanceof Error) {
         throw new CredentialFetchingError(
-          `fetching credentials: ${(error as Error).message}`
+          `fetching credentials: ${error.name}: ${error.message}`
         )
       }
+      throw error
     }
   }
 
   async reportJobError(error: JobError): Promise<void> {
-    const recordErrorURL = `/update_jobs/${this.params.jobId}/record_update_job_error`
-    const res = await this.client.post(
+    const recordErrorURL = `${this.params.dependabotApiUrl}/update_jobs/${this.params.jobId}/record_update_job_error`
+    const res = await this.client.postJson(
       recordErrorURL,
       {data: error},
       {
-        headers: {Authorization: this.params.jobToken}
+        ['Authorization']: this.params.jobToken
       }
     )
-    if (res.status !== 204) {
-      throw new Error(`Unexpected status code: ${res.status}`)
+    if (res.statusCode !== 204) {
+      throw new Error(`Unexpected status code: ${res.statusCode}`)
     }
   }
 
   async markJobAsProcessed(): Promise<void> {
-    const markAsProcessedURL = `/update_jobs/${this.params.jobId}/mark_as_processed`
-    const res = await this.client.patch(
+    const markAsProcessedURL = `${this.params.dependabotApiUrl}/update_jobs/${this.params.jobId}/mark_as_processed`
+    const res = await this.client.patchJson(
       markAsProcessedURL,
       {data: this.UnknownSha},
       {
-        headers: {Authorization: this.params.jobToken}
+        ['Authorization']: this.params.jobToken
       }
     )
-    if (res.status !== 204) {
-      throw new Error(`Unexpected status code: ${res.status}`)
+    if (res.statusCode !== 204) {
+      throw new Error(`Unexpected status code: ${res.statusCode}`)
+    }
+  }
+
+  private async getJsonWithRetry<T>(
+    url: string,
+    token: string
+  ): Promise<TypedResponse<T>> {
+    let attempt = 1
+
+    const execute = async (): Promise<TypedResponse<T>> => {
```

Comment:
how does the actions team handle retries? Do they have some utility we can reuse?

Reply:
It's embedded in the library and not exported: https://github.com/actions/toolkit/blob/fe3e7ce9a7f995d29d1fcfd226a32bca407f9dc8/packages/http-client/src/index.ts#L444-L450